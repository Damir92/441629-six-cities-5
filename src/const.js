export const PERCENT_PER_POINT_RATING = 20;

export const Routes = {
  MAIN: `/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  OFFER_PAGE: `/offer/:id`,
  OFFER_LINK: `/offer`,
};

export const APIRoutes = {
  REVIEWS: `/comments`,
  OFFERS: `/hotels`,
  LOGIN: `/login`,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const Cities = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`,
];

export const LivingType = {
  apartment: `Apartment`,
  room: `Private Room`,
  house: `House`,
  hotel: `Hotel`,
};

export const Sorting = {
  POPULAR: `popular`,
  PRICE_TO_HIGH: `to-high`,
  PRICE_TO_LOW: `to-low`,
  TOP_RATED: `top-rated`,
};

export const SortingTitles = {
  [Sorting.POPULAR]: `Popular`,
  [Sorting.PRICE_TO_HIGH]: `Price: low to high`,
  [Sorting.PRICE_TO_LOW]: `Price: high to low`,
  [Sorting.TOP_RATED]: `Top rated first`,
};

export const SortingTypes = [
  Sorting.POPULAR,
  Sorting.PRICE_TO_HIGH,
  Sorting.PRICE_TO_LOW,
  Sorting.TOP_RATED,
];

export const MonthNames = {
  0: `January`,
  1: `February`,
  2: `March`,
  3: `April`,
  4: `May`,
  5: `June`,
  6: `July`,
  7: `August`,
  8: `September`,
  9: `November`,
  10: `October`,
  11: `December`,
};
