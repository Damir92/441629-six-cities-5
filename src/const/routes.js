const ROUTES = {
  MAIN: `/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  OFFER_PAGE: `/offer/:id`,
};

export {ROUTES};
