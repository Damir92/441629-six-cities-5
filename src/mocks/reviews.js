export const reviews = [
  {
    id: 1,
    avatar: `img/avatar-max.jpg`,
    date: `2019-04-24`,
    dateHuman: `April 2019`,
    name: `Max`,
    rating: 4,
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
  },
  {
    id: 2,
    avatar: `img/avatar-max.jpg`,
    date: `2019-04-24`,
    dateHuman: `April 2019`,
    name: `Max`,
    rating: 4,
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
  },
];
