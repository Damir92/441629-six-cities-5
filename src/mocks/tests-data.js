export const cityOffers = [
  {
    bedrooms: 1,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
    },
    description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
    goods: [
      `Fridge`,
      `Washer`,
      `Coffee machine`,
      `Washing machine`,
      `Dishwasher`,
      `Laptop friendly workspace`,
      `Cable TV`,
      `Air conditioning`,
      `Baby seat`,
      `Towels`,
      `Breakfast`,
    ],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isPro: true,
      name: `Angelina`,
    },
    id: 2,
    images: [
      `imgURL-1`,
      `imgURL-2`,
    ],
    isFavorite: false,
    isPremium: false,
    location: {
      latitude: 52.37454,
      longitude: 4.881976,
      zoom: 16,
    },
    volume: 1,
    preview: `previewURL`,
    price: 111,
    rating: 3.3,
    title: `Wood and stone place`,
    type: `room`,
  },
  {
    bedrooms: 1,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
    },
    description: `A new spacious villa, one floor. All commodities, jacuzzi and beautiful scenery. Ideal for families or friends.`,
    goods: [
      `Laptop friendly workspace`,
      `Baby seat`,
      `Towels`,
      `Air conditioning`,
      `Breakfast`,
      `Washer`,
    ],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isPro: true,
      name: `Angelina`,
    },
    id: 4,
    images: [
      `imgURL-1`,
      `imgURL-2`,
    ],
    isFavorite: true,
    isPremium: false,
    location: {
      latitude: 52.35054,
      longitude: 4.908976,
      zoom: 16,
    },
    volume: 2,
    preview: `previewURL`,
    price: 294,
    rating: 3.8,
    title: `Loft Studio in the Central Area`,
    type: `hotel`,
  }, {
    bedrooms: 4,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
    },
    description: `This is a place for dreamers to reset, reflect, and create. Designed with a "slow" pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
    goods: [
      `Air conditioning`,
      `Baby seat`,
      `Fridge`,
      `Towels`,
      `Breakfast`,
      `Washer`,
      `Laptop friendly workspace`,
    ],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isPro: true,
      name: `Angelina`,
    },
    id: 9,
    images: [
      `imgURL-1`,
      `imgURL-2`,
    ],
    isFavorite: false,
    isPremium: false,
    location: {
      latitude: 52.37854,
      longitude: 4.894976,
      zoom: 16,
    },
    volume: 10,
    preview: `previewURL`,
    price: 406,
    rating: 2.2,
    title: `Loft Studio in the Central Area`,
    type: `hotel`,
  },
];
