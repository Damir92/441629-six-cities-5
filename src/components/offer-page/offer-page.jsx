import React, {useEffect} from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

import {OfferPropTypes, OfferPagePropTypes, ReviewPropTypes} from '../../prop-types';
import {AuthorizationStatus, FavoriteStatus, LivingType, PageTypes, Routes} from '../../const';
import {convertRatingToPercent} from '../../utils';

import {getActiveOffer, getReviews, getNearbyOffers} from '../../store/reducers/site-data/selectors';
import {getAuthorizationStatus} from '../../store/reducers/user/selectors';
import {fetchActiveOffer, fetchReviews, postReview, fetchNearbyOffers, updateFavorite} from '../../store/api-actions';
import {unsetActiveOfferAction} from '../../store/action';

import Header from '../header/header';
import ReviewsList from '../reviews-list/reviews-list';
import ReviewForm from '../review-form/review-form';
import OffersList from '../offers-list/offers-list';
import OffersMap from '../offers-map/offers-map';

const OfferPage = (props) => {
  const {
    nearbyOffers,
    logged,
    history,
    match,
    offer = {},
    onFavoriteClick,
    onLoad,
    onSendForm,
    reviews = [],
    unsetActiveOffer,
  } = props;

  const offerId = +match.params.id;

  useEffect(() => () => unsetActiveOffer(), []);

  useEffect(() => onLoad(offerId), [match.params.id]);

  const {
    bedrooms,
    description,
    goods,
    host,
    images,
    isFavorite,
    isPremium,
    volume,
    price,
    rating,
    title,
    type,
  } = offer;

  const handleFavoriteClick = () => {
    if (logged === AuthorizationStatus.NO_AUTH) {
      history.push(Routes.LOGIN);
    } else {
      onFavoriteClick({
        id: offerId,
        status: isFavorite ? FavoriteStatus.IS_NOT_FAVORITE : FavoriteStatus.IS_FAVORITE,
      });
    }
  };

  return (
    <div className="page">

      <Header />

      <main className="page__main page__main--property">
        <section className="property">
          <div className="property__gallery-container container">
            <div className="property__gallery">
              {images && images.slice(0, 6).map((image) => (
                <div
                  key={image}
                  className="property__image-wrapper"
                >
                  <img className="property__image" src={image} alt="Photo studio" />
                </div>
              ))}
            </div>
          </div>
          <div className="property__container container">
            <div className="property__wrapper">
              {
                isPremium
                &&
                <div className="property__mark">
                  <span>Premium</span>
                </div>
              }
              <div className="property__name-wrapper">
                <h1 className="property__name">
                  {title}
                </h1>
                <button
                  className={`property__bookmark-button button ${isFavorite ? `property__bookmark-button--active` : ``}`}
                  type="button"
                  onClick={handleFavoriteClick}
                >
                  <svg className="property__bookmark-icon place-card__bookmark-icon" width="31" height="33">
                    <use xlinkHref="#icon-bookmark"></use>
                  </svg>
                  <span className="visually-hidden">
                    {isFavorite ? `In bookmarks` : `To bookmarks`}
                  </span>
                </button>
              </div>
              <div className="property__rating rating">
                <div className="property__stars rating__stars">
                  <span style={{width: convertRatingToPercent(rating)}}></span>
                  <span className="visually-hidden">Rating</span>
                </div>
                <span className="property__rating-value rating__value">{rating}</span>
              </div>
              <ul className="property__features">
                <li className="property__feature property__feature--entire">
                  {LivingType[type]}
                </li>
                <li className="property__feature property__feature--bedrooms">
                  {bedrooms} Bedrooms
                </li>
                <li className="property__feature property__feature--adults">
                  Max {volume} adults
                </li>
              </ul>
              <div className="property__price">
                <b className="property__price-value">&euro;{price}</b>
                <span className="property__price-text">&nbsp;night</span>
              </div>
              <div className="property__inside">
                <h2 className="property__inside-title">What&apos;s inside</h2>
                <ul className="property__inside-list">
                  {goods && goods.map((item) => (
                    <li
                      key={item}
                      className="property__inside-item"
                    >
                      {item}
                    </li>
                  ))}
                </ul>
              </div>
              <div className="property__host">
                <h2 className="property__host-title">Meet the host</h2>
                <div className="property__host-user user">
                  <div className={`property__avatar-wrapper user__avatar-wrapper ${host && host.isPro ? `property__avatar-wrapper--pro` : ``}`}>
                    <img className="property__avatar user__avatar" src={host && host.avatar} width="74" height="74" alt="Host avatar" />
                  </div>
                  <span className="property__user-name">
                    {host && host.name}
                  </span>
                </div>
                <div className="property__description">
                  <p className="property__text">
                    {description}
                  </p>
                </div>
              </div>
              <section className="property__reviews reviews">
                <h2 className="reviews__title">Reviews &middot; <span className="reviews__amount">{reviews.length || 0}</span></h2>

                <ReviewsList
                  reviews={reviews}
                />

                {logged === AuthorizationStatus.AUTH
                  ?
                  <ReviewForm
                    id={offerId}
                    onSubmit={onSendForm}
                  />
                  :
                  null
                }

              </section>
            </div>
          </div>

          <section className="property__map map">

            {nearbyOffers
              ?
              <OffersMap
                activeOffer={offer}
                offers={nearbyOffers}
              />
              :
              null}

          </section>

        </section>
        <div className="container">
          <section className="near-places places">
            <h2 className="near-places__title">Other places in the neighbourhood</h2>

            {nearbyOffers
              ?
              <OffersList
                cityOffers={nearbyOffers}
                pageType={PageTypes.OFFER}
              />
              :
              null}

          </section>
        </div>
      </main>
    </div>
  );
};

OfferPage.propTypes = {
  offer: PropTypes.oneOfType([
    PropTypes.shape(OfferPagePropTypes).isRequired,
    PropTypes.shape({}).isRequired
  ]).isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
  match: PropTypes.object.isRequired,
  nearbyOffers: PropTypes.arrayOf(
      PropTypes.shape(OfferPropTypes).isRequired
  ).isRequired,
  logged: PropTypes.oneOf([AuthorizationStatus.AUTH, AuthorizationStatus.NO_AUTH]).isRequired,
  onFavoriteClick: PropTypes.func.isRequired,
  onLoad: PropTypes.func.isRequired,
  onSendForm: PropTypes.func.isRequired,
  reviews: PropTypes.arrayOf(
      PropTypes.shape(ReviewPropTypes)
  ).isRequired,
  unsetActiveOffer: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  logged: getAuthorizationStatus(state),
  nearbyOffers: getNearbyOffers(state),
  offer: getActiveOffer(state),
  reviews: getReviews(state),
});

const mapDispatchToProps = (dispatch) => ({
  onLoad: (id) => {
    dispatch(fetchActiveOffer(id));
    dispatch(fetchReviews(id));
    dispatch(fetchNearbyOffers(id));
  },
  onFavoriteClick: (value) => dispatch(updateFavorite(value)),
  onSendForm: (reviewData) => dispatch(postReview(reviewData)),
  unsetActiveOffer: () => dispatch(unsetActiveOfferAction()),
});

export {OfferPage};
export default connect(mapStateToProps, mapDispatchToProps)(OfferPage);
