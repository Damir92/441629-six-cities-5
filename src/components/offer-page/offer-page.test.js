import React from 'react';
import {shallow} from 'enzyme';

import {cityOffers, offer, noop, reviews} from '../../mocks/tests-data';
import {AuthorizationStatus} from '../../const';

import {OfferPage} from './offer-page';

const match = {
  params: {
    id: 0,
  },
};

describe(`Offer page renders correctly`, () => {
  it(`It is not main page`, () => {
    const tree = shallow(
        <OfferPage
          cityOffers={cityOffers}
          logged={AuthorizationStatus.NO_AUTH}
          match={match}
          offer={offer}
          onLoad={noop}
          onSendForm={noop}
          reviews={reviews}
          unloadActiveOffer={noop}
          userData={null}
          unsetActiveOffer={noop}
        />
    );

    expect(tree).toMatchSnapshot();
  });
});
