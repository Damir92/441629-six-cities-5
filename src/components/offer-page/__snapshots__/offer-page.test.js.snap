// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offer page renders correctly It is not main page 1`] = `
<div
  className="page"
>
  <Connect(Header)
    history={
      Object {
        "push": [Function],
      }
    }
  />
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
            key="imgURL-1"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="imgURL-1"
            />
          </div>
          <div
            className="property__image-wrapper"
            key="imgURL-2"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="imgURL-2"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Loft Studio in the Central Area
            </h1>
            <button
              className="property__bookmark-button button "
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "40%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              2.2
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              Hotel
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              4
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              10
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              406
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
                key="Air conditioning"
              >
                Air conditioning
              </li>
              <li
                className="property__inside-item"
                key="Baby seat"
              >
                Baby seat
              </li>
              <li
                className="property__inside-item"
                key="Fridge"
              >
                Fridge
              </li>
              <li
                className="property__inside-item"
                key="Towels"
              >
                Towels
              </li>
              <li
                className="property__inside-item"
                key="Breakfast"
              >
                Breakfast
              </li>
              <li
                className="property__inside-item"
                key="Washer"
              >
                Washer
              </li>
              <li
                className="property__inside-item"
                key="Laptop friendly workspace"
              >
                Laptop friendly workspace
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper user__avatar-wrapper "
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                This is a place for dreamers to reset, reflect, and create. Designed with a "slow" pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews · 
              <span
                className="reviews__amount"
              >
                2
              </span>
            </h2>
            <ReviewsList
              reviews={
                Array [
                  Object {
                    "avatar": "img/avatar-max.jpg",
                    "date": "2019-04-24",
                    "dateHuman": "April 2019",
                    "id": 1,
                    "name": "Max",
                    "rating": 4,
                    "text": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                  },
                  Object {
                    "avatar": "img/avatar-max.jpg",
                    "date": "2019-04-24",
                    "dateHuman": "April 2019",
                    "id": 2,
                    "name": "Max",
                    "rating": 4,
                    "text": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                  },
                ]
              }
            />
          </section>
        </div>
      </div>
      <section
        className="property__map map"
      >
        <Connect(OffersMap)
          cityOffers={
            Array [
              Object {
                "bedrooms": 1,
                "city": Object {
                  "location": Object {
                    "latitude": 52.37454,
                    "longitude": 4.897976,
                    "zoom": 13,
                  },
                  "name": "Amsterdam",
                },
                "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
                "goods": Array [
                  "Fridge",
                  "Washer",
                  "Coffee machine",
                  "Washing machine",
                  "Dishwasher",
                  "Laptop friendly workspace",
                  "Cable TV",
                  "Air conditioning",
                  "Baby seat",
                  "Towels",
                  "Breakfast",
                ],
                "host": Object {
                  "avatar": "img/avatar-angelina.jpg",
                  "id": 25,
                  "isPro": true,
                  "name": "Angelina",
                },
                "id": 2,
                "images": Array [
                  "imgURL-1",
                  "imgURL-2",
                ],
                "isFavorite": false,
                "isPremium": false,
                "location": Object {
                  "latitude": 52.37454,
                  "longitude": 4.881976,
                  "zoom": 16,
                },
                "preview": "previewURL",
                "price": 111,
                "rating": 3.3,
                "title": "Wood and stone place",
                "type": "room",
                "volume": 1,
              },
              Object {
                "bedrooms": 1,
                "city": Object {
                  "location": Object {
                    "latitude": 52.37454,
                    "longitude": 4.897976,
                    "zoom": 13,
                  },
                  "name": "Amsterdam",
                },
                "description": "A new spacious villa, one floor. All commodities, jacuzzi and beautiful scenery. Ideal for families or friends.",
                "goods": Array [
                  "Laptop friendly workspace",
                  "Baby seat",
                  "Towels",
                  "Air conditioning",
                  "Breakfast",
                  "Washer",
                ],
                "host": Object {
                  "avatar": "img/avatar-angelina.jpg",
                  "id": 25,
                  "isPro": true,
                  "name": "Angelina",
                },
                "id": 4,
                "images": Array [
                  "imgURL-1",
                  "imgURL-2",
                ],
                "isFavorite": true,
                "isPremium": false,
                "location": Object {
                  "latitude": 52.35054,
                  "longitude": 4.908976,
                  "zoom": 16,
                },
                "preview": "previewURL",
                "price": 294,
                "rating": 3.8,
                "title": "Loft Studio in the Central Area",
                "type": "hotel",
                "volume": 2,
              },
              Object {
                "bedrooms": 4,
                "city": Object {
                  "location": Object {
                    "latitude": 52.37454,
                    "longitude": 4.897976,
                    "zoom": 13,
                  },
                  "name": "Amsterdam",
                },
                "description": "This is a place for dreamers to reset, reflect, and create. Designed with a \\"slow\\" pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.",
                "goods": Array [
                  "Air conditioning",
                  "Baby seat",
                  "Fridge",
                  "Towels",
                  "Breakfast",
                  "Washer",
                  "Laptop friendly workspace",
                ],
                "host": Object {
                  "avatar": "img/avatar-angelina.jpg",
                  "id": 25,
                  "isPro": true,
                  "name": "Angelina",
                },
                "id": 9,
                "images": Array [
                  "imgURL-1",
                  "imgURL-2",
                ],
                "isFavorite": false,
                "isPremium": false,
                "location": Object {
                  "latitude": 52.37854,
                  "longitude": 4.894976,
                  "zoom": 16,
                },
                "preview": "previewURL",
                "price": 406,
                "rating": 2.2,
                "title": "Loft Studio in the Central Area",
                "type": "hotel",
                "volume": 10,
              },
            ]
          }
        />
      </section>
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <OffersList
          cityOffers={
            Array [
              Object {
                "bedrooms": 1,
                "city": Object {
                  "location": Object {
                    "latitude": 52.37454,
                    "longitude": 4.897976,
                    "zoom": 13,
                  },
                  "name": "Amsterdam",
                },
                "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
                "goods": Array [
                  "Fridge",
                  "Washer",
                  "Coffee machine",
                  "Washing machine",
                  "Dishwasher",
                  "Laptop friendly workspace",
                  "Cable TV",
                  "Air conditioning",
                  "Baby seat",
                  "Towels",
                  "Breakfast",
                ],
                "host": Object {
                  "avatar": "img/avatar-angelina.jpg",
                  "id": 25,
                  "isPro": true,
                  "name": "Angelina",
                },
                "id": 2,
                "images": Array [
                  "imgURL-1",
                  "imgURL-2",
                ],
                "isFavorite": false,
                "isPremium": false,
                "location": Object {
                  "latitude": 52.37454,
                  "longitude": 4.881976,
                  "zoom": 16,
                },
                "preview": "previewURL",
                "price": 111,
                "rating": 3.3,
                "title": "Wood and stone place",
                "type": "room",
                "volume": 1,
              },
              Object {
                "bedrooms": 1,
                "city": Object {
                  "location": Object {
                    "latitude": 52.37454,
                    "longitude": 4.897976,
                    "zoom": 13,
                  },
                  "name": "Amsterdam",
                },
                "description": "A new spacious villa, one floor. All commodities, jacuzzi and beautiful scenery. Ideal for families or friends.",
                "goods": Array [
                  "Laptop friendly workspace",
                  "Baby seat",
                  "Towels",
                  "Air conditioning",
                  "Breakfast",
                  "Washer",
                ],
                "host": Object {
                  "avatar": "img/avatar-angelina.jpg",
                  "id": 25,
                  "isPro": true,
                  "name": "Angelina",
                },
                "id": 4,
                "images": Array [
                  "imgURL-1",
                  "imgURL-2",
                ],
                "isFavorite": true,
                "isPremium": false,
                "location": Object {
                  "latitude": 52.35054,
                  "longitude": 4.908976,
                  "zoom": 16,
                },
                "preview": "previewURL",
                "price": 294,
                "rating": 3.8,
                "title": "Loft Studio in the Central Area",
                "type": "hotel",
                "volume": 2,
              },
              Object {
                "bedrooms": 4,
                "city": Object {
                  "location": Object {
                    "latitude": 52.37454,
                    "longitude": 4.897976,
                    "zoom": 13,
                  },
                  "name": "Amsterdam",
                },
                "description": "This is a place for dreamers to reset, reflect, and create. Designed with a \\"slow\\" pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.",
                "goods": Array [
                  "Air conditioning",
                  "Baby seat",
                  "Fridge",
                  "Towels",
                  "Breakfast",
                  "Washer",
                  "Laptop friendly workspace",
                ],
                "host": Object {
                  "avatar": "img/avatar-angelina.jpg",
                  "id": 25,
                  "isPro": true,
                  "name": "Angelina",
                },
                "id": 9,
                "images": Array [
                  "imgURL-1",
                  "imgURL-2",
                ],
                "isFavorite": false,
                "isPremium": false,
                "location": Object {
                  "latitude": 52.37854,
                  "longitude": 4.894976,
                  "zoom": 16,
                },
                "preview": "previewURL",
                "price": 406,
                "rating": 2.2,
                "title": "Loft Studio in the Central Area",
                "type": "hotel",
                "volume": 10,
              },
            ]
          }
          history={
            Object {
              "push": [Function],
            }
          }
          isMainPage={false}
        />
      </section>
    </div>
  </main>
</div>
`;
